src/main/java/backend/resource/TurboMilestoneDueDateComparator.java

```diff
@@ -0,0 +1,27 @@
+package backend.resource;
+
+
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.util.Comparator;
+
+public class TurboMilestoneDueDateComparator implements Comparator<TurboMilestone>,
+        Serializable {
+
+    @Override
+    public int compare(TurboMilestone m1, TurboMilestone m2) {
+        if (!m1.getDueDate().isPresent() && !m2.getDueDate().isPresent()) {
+            return 0;
+        } else if (!m1.getDueDate().isPresent()) {
+            return 1;
+        } else if (!m2.getDueDate().isPresent()) {
+            return -1;
+        } else {
+            LocalDate m1DueDate = m1.getDueDate().get();
+            LocalDate m2DueDate = m2.getDueDate().get();
+
+            return (m1DueDate.isBefore(m2DueDate)) ? -1 :
+                    (m1DueDate.isEqual(m2DueDate)) ? 0 : 1;
+        }
+    }
```

Comment:
Which one is better? As a new file or method? I think method is cleaner
since I put the file in the same folder with `TurboMilestone` right now but
it seems to be out of place.
On Oct 12, 2015 10:11 PM, "Darius Foo" <notifications@github.com> wrote:

> In src/main/java/backend/resource/TurboMilestoneDueDateComparator.java
> <https://github.com/HubTurbo/HubTurbo/pull/1050#discussion_r41759652>:
>
> > +    @Override
> > +    public int compare(TurboMilestone m1, TurboMilestone m2) {
> > +        if (!m1.getDueDate().isPresent() && !m2.getDueDate().isPresent()) {
> > +            return 0;
> > +        } else if (!m1.getDueDate().isPresent()) {
> > +            return 1;
> > +        } else if (!m2.getDueDate().isPresent()) {
> > +            return -1;
> > +        } else {
> > +            LocalDate m1DueDate = m1.getDueDate().get();
> > +            LocalDate m2DueDate = m2.getDueDate().get();
> > +
> > +            return (m1DueDate.isBefore(m2DueDate)) ? -1 :
> > +                    (m1DueDate.isEqual(m2DueDate)) ? 0 : 1;
> > +        }
> > +    }
>
> A method would be fine, which you can refer to with this::comparator. But
> yeah, we can only resolve in favour of one of your implementations.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/HubTurbo/HubTurbo/pull/1050/files#r41759652>.
>